lib/fs.js

```diff
@@ -550,9 +553,17 @@ function read(fd, buffer, offset, length, position, callback) {
 
   validateOffsetLengthRead(offset, length, buffer.byteLength);
 
-  if (!NumberIsSafeInteger(position))
+  if (position == null)
     position = -1;
 
+  if (!(NumberIsSafeInteger(position) || typeof position === 'bigint'))
```

Comment:
Can you split the checks into

1. If it's a number and a safe number
2. If it's a bigint and in a safe range

And avoid converting the parameter to BigInt when we are sure it's a number?

Reply:
Updated :)
I was thinking if I should also add a `validateInt64` to `internal/validators.js` because that might get rid of some repetitions in the validation.